app/api/appointments/getAppointments.ts

```diff
@@ -102,7 +102,37 @@ export async function getAvailableTimeSlots(jobType: JobType): Promise<DateEntry
         }
         const appointmentsResponse = await serviceTitanClient.jpm.AppointmentsService.appointmentsGetList(appointmentsQuery);
         const appointments = appointmentsResponse.data;
-        logger.info(`Found ${appointments.length} appointments for technician ${tech.technicianName}`);
+        // logger.info(`Found ${appointments.length} appointments for technician ${tech.technicianName}`);
+
+        // Filter out cancelled appointments by checking the associated job status
+        const activeAppointments: Jpm_V2_AppointmentResponse[] = [];
+
+        let cancelledCount = 0;
+
+        // Cache job statuses to minimize API calls
+        const jobStatusCache = new Map<number, string>();
+        
+        for (const appointment of appointments) {
+            let jobStatus = jobStatusCache.get(appointment.jobId);
+            if (!jobStatus) {
+                const job = await serviceTitanClient.jpm.JobsService.jobsGet({
```

Comment:
how did we test this? In looking at the API (here: https://developer.servicetitan.io/api-details/#api=tenant-jpm-v2&operation=Appointments_GetList) it looks like appointments do have a status.... the question is: When I cancel a Job will the Appointment automattically get cancelled as well?
Because if so, then we don't need to do the additional `jobsGet` query here.

If you haven't tested it, you can:
1. Use the app to create a job
2. Use a direct api call to cancel the job Ex. `https://developer.servicetitan.io/api-details/#api=tenant-jpm-v2&operation=Jobs_Cancel`
3. Use the app to reload the schedule screen and check logs to verify that the timeslot is now open again.